src/pages/index.js

```diff
@@ -2,34 +2,116 @@ import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import "../pages/index.css";
 import {
-  initialCards,
   cardContainer,
   cardSelector,
   cardForm,
   settings,
   buttonEdit,
-  titleInput,
   buttonAdd,
-  linkInput,
   nameInput,
   jobInput,
+  changeAvatarbtn,
+  apiOptions,
+  avatarInput,
+  savingBtn,
 } from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-
+import Api from "../components/Api.js";
+// import Popup from "../components/Popup.js";
+import PopupConfirmDelete from "../components/PopupConfirmDelete.js";
 /*-----------------------------------------------------------------*/
 /*                           Functions                             */
 /*-----------------------------------------------------------------*/
 
 function createCard(data) {
-  const cardElement = new Card(data, cardSelector, (imgData) => {
-    imagePopup.open(imgData);
-  });
+  const cardElement = new Card(
+    data,
+    cardSelector,
+    (imgData) => {
+      imagePopup.open(imgData);
+    },
+    handleCardDelete,
+    handleCardLike
+  );
   return cardElement.cardView();
 }
 
+// Api instances
+const api = new Api(apiOptions);
+
+// delete card instance
+function handleCardDelete(card) {
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        //remove card from DOM
+        card.deleteCard();
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card", err);
+      });
+  });
+  confirmDeleteModal.open();
+}
+
+// Confirm delete modal instance
+const confirmDeleteModal = new PopupConfirmDelete("#confirm-delete-modal");
+
+confirmDeleteModal.setEventListeners();
+
+//Cards should be rendered after the user information is received from the server.
+
+function getUserData() {
+  api
+    .getUserInfo()
+    .then((data) => {
+      userInfo.setUserInfo(data);
+    })
+    .catch((err) => console.error(err));
+}
+getUserData();
```

Comment:
`getUserData` is not needed because you get the data in `Promise.all` in line 85